src/bcc2.c

```diff
@@ -39,6 +43,26 @@ parse_args(int argc, char *argv[]) {
       flags.reg_dump |= strcmp(argv[i], "-regs") == 0;
       flags.help |= strcmp(argv[i], "-h") == 0;
       flags.version |= strcmp(argv[i], "-v") == 0;
+
+      if (strcmp(argv[i], "-platform") == 0) {
+        if (i + 1 >= argc) {
+          flags.list_platforms |= 1;
+        } else if (argv[i + 1][0] == '-') {
+          log_err_final("expected platform name after -platform");
```

Comment:
Shouldn't this be the same behaviour as the if statement above? Why should the platforms only be listed if -platforms is the last item in the args?